services/QuillLMS/app/models/concerns/vitally_school_stats.rb

```diff
@@ -22,15 +22,20 @@ module VitallySchoolStats
       .joins("JOIN classrooms ON classrooms.id=classrooms_teachers.classroom_id")
       .joins("JOIN classroom_units ON classroom_units.classroom_id=classrooms.id")
       .joins("JOIN activity_sessions ON activity_sessions.classroom_unit_id=classroom_units.id")
+      .joins("JOIN activities ON activity_sessions.activity_id = activities.id")
       .where('schools.id = ?', school.id)
       .where('activity_sessions.state = ?', 'finished')
   end
 
-  def activities_per_student(active_students, activities_finished)
-    if active_students > 0
-      (activities_finished.to_f / active_students).round(2)
-    else
-      0
-    end
+  def activities_assigned_query(school)
+    ClassroomUnit.joins("JOIN unit_activities ON classroom_units.unit_id=unit_activities.unit_id")
+    .joins("JOIN activities ON activities.id = unit_activities.activity_id")
+    .joins("JOIN classrooms ON classrooms.id = classroom_units.classroom_id")
+    .joins("JOIN classrooms_teachers ON classrooms.id=classrooms_teachers.classroom_id")
+    .joins("JOIN schools_users ON schools_users.user_id = classrooms_teachers.user_id")
+    .joins("JOIN schools ON schools_users.school_id=schools.id")
+    .where("schools.id = ?", school.id)
+    .select("assigned_student_ids, activities.id, unit_activities.created_at")
```

Comment:
I don't know that this is better, but we can take advantage of all the `has_many` and other relationships that we've put into our model when making our join calls.  This should result in the same SQL, so you can just skip this, but I figured I'd call it out.:
```ruby
ClassroomUnit.joins(classroom: {teachers: :school}, unit: :activities)
  .where("schools.id = ?", school.id)
  .select("assigned_student_ids", "activities.id", "unit_activities.created_at")
```